Clean up Helix.props and Helix.Common.props

dougbu opened the pull request:
- address confusion hit when fixing an internal 6.0 PR
- generally, reflect Helix.props' use of project settings better
  - checks and settings that don't use project settings are in Helix.Common.props
- move `$(Is...Queue)` settings to Helix.Common.props
  - these properties depend on the current queue and that's known in helix.proj
- move `$(IsWindowsOnlyTest)` checks to Helix.props
  - property is set in projects; never seen in helix.proj

nit: also move `$(RunQuarantinedTests)` default to Helix.Common.props
  - global property is visible in helix.proj and may at some point be useful there

dougbu committed:
Clean up Helix.props and Helix.Common.props
- address confusion hit when fixing an internal 6.0 PR
- generally, reflect Helix.props' use of project settings better
  - checks and settings that don't use project settings are in Helix.Common.props
- move `$(Is...Queue)` settings to Helix.Common.props
  - these properties depend on the current queue and that's known in helix.proj
- move `$(IsWindowsOnlyTest)` checks to Helix.props
  - property is set in projects; never seen in helix.proj

nit: also move `$(RunQuarantinedTests)` default to Helix.Common.props
  - global property is visible in helix.proj and may at some point be useful there

dougbu commented:
@**reviewers** I would also appreciate your thoughts on backporting this or something like it to release/6.0. The change here is a bit clearer than what we did in internal/release/6.0 and that will take a while to show up outside that branch.

wtgodbe reviewed (approved):
I'm on board with backporting this to 6.0 when we clean up the workarounds from the internal PR

dougbu commented on eng/targets/Helix.Common.props:
```diff
@@ -6,17 +6,16 @@
   </PropertyGroup>
 
-  <!-- Don't use <Choose/> for $(IsWindowsOnlyTest) checks. <When/> Condition is evaluated before project load. -->
-  <ItemGroup Condition=" '$(IsWindowsOnlyTest)' != 'true' ">
```
To be clear: This worked but only because item group conditions are evaluated after the `msbuild` pass that sets all properties. It never worked in helix.proj and the item group is only used in Helix.targets.

I could have moved this `<ItemGroup/>` to Helix.props but decided to leave that file w/ only the bits that could only matter w/in our .csproj files or Helix.targets. `@(HelixAvailablePlatform)` is a fairly generic item group.

HaoK commented on eng/targets/Helix.props:
```diff
@@ -10,33 +10,20 @@
-  </PropertyGroup>
-
-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+  <ItemGroup Condition=" '$(TargetFrameworkIdentifier)' == '.NETFramework' OR '$(IsWindowsOnlyTest)' == 'true' ">
```
I'm guessing there's no way to cleanly do an if/else type of thing, since I'm reading these two item groups as basically an if/else right?

dougbu replied:
`<Choose/>` provides the main `if`/`else` option in `msbuild` but `<Choose/>` conditions get evaluated a bit too early to make their use simple for properties changed in TFM-specific project builds. I guess we could move this to Helix.targets but I'm less worried about that than the previous confusion between the two .props files.

dougbu replied:
Made the change to Helix.targets and `<Choose/>` in second commit (along w/ fixing Helix.Common.props).

dougbu committed:
!fixup! Fix build and address review comments
- correct how `$(Is...Queue)` properties are set
- move `@(HelixProjectPlatform)` settings to Helix.targets
  - use `<Choose/>`
